src/corva/service/cache_sdk.py

```diff
@@ -0,0 +1,159 @@
+import datetime
+import warnings
+from typing import List, Optional, Protocol, overload
+
+import fakeredis
+import redis
+
+from corva import cache_adapter
+
+
+class UserCacheSdkProtocol(Protocol):
+    def set(self, key: str, value: str, ttl: int = ...) -> None:
+        ...
+
+    def get(self, key: str) -> Optional[str]:
+        ...
+
+    # TODO: remove asterisk in v2 - it was added for backward compatibility
+    def delete(self, *, key: str) -> None:
+        ...
+
+
+class UserRedisSdk:
+    """User cache protocol implementation using Redis.
+
+    As AWS Lambda is meant to be stateless, the apps need some mechanism to share the
+    data between invokes. Redis provides an in-memory low latency storage for such data.
+    """
+
+    SIXTY_DAYS: int = int(datetime.timedelta(days=60).total_seconds())
+
+    def __init__(self, hash_name: str, redis_dsn: str, use_fakes: bool = False):
+        if use_fakes:
+            client = fakeredis.FakeRedis.from_url(url=redis_dsn, decode_responses=True)
+        else:
+            client = redis.Redis.from_url(url=redis_dsn, decode_responses=True)
+
+        self.cache_repo = cache_adapter.RedisRepository(
+            hash_name=hash_name, client=client
+        )
+        self.old_cache_repo = cache_adapter.DeprecatedRedisAdapter(
+            hash_name=hash_name, client=client
+        )
+
+    def set(self, key: str, value: str, ttl: int = SIXTY_DAYS) -> None:
+        self.cache_repo.set(key=key, value=value, ttl=ttl)
+
+    def get(self, key: str) -> Optional[str]:
+        return self.cache_repo.get(key=key)
+
+    @overload
+    def delete(self, *, key: str) -> None:
+        ...
+
+    @overload
+    def delete(self, keys: List[str], name: Optional[str] = ...) -> int:
+        ...
+
+    def delete(
+        self,
+        keys=None,
+        name=None,
+        *,
+        key=None,
+    ):
+        if keys is not None:
+            warnings.warn(
```

Comment:
Yes.

> [Warning messages are normally written to sys.stderr](https://docs.python.org/3.8/library/warnings.html#:~:text=Warning%20messages%20are%20normally%20written%20to%20sys.stderr)

Reply:
Doesn't this cause excessive logging for customers who update SDK but don't convert old cache calls?

Reply:
It will be printed only once for each location (module + line number). So should not be too excessive

https://docs.python.org/3.8/library/warnings.html#:~:text=Disposition-,%22default%22,-print%20the%20first

```python

import warnings


def func():
    warnings.warn("Hello", FutureWarning)


func()
func()

# > FutureWarning: Hello
# >  warnings.warn("Hello", FutureWarning)

```